crates/controller/src/hud/map.rs

```diff
@@ -0,0 +1,30 @@
+use bevy::prelude::{default, AlignSelf, Color, Component, NodeBundle, Size, Style, Val};
+
+pub(crate) struct MapBundle {
```

Comment:
* It is not a bundle, since you don't directly spawn it with Bevy.
* Having a struct like this is overkill, it can be a function.
* Further: I would construct & spawn the node directly in `        .with_children(|parent| {`

Reply:
Making it a function seems reasonable, but as of my vision... I thought about it as an **object made of two parts**.. **First**, there is (possibly) **a static part**, that should be drawn, like terrain, at the moment this part is drawn as a `color`. The **second** part should be **dynamic** and it should be made of small dots or whatever the design, and moved around the same as an object.. When _the object_ is moving, this same action should trigger system that is projected on a mini-map  to move a dot around as well..

Reply:
It can be easily changed once there is something drawn to the rect. We need to research how to do it and the final implemented design will almost certainly differ from what you have described (so this code won't survive). Thus I would stick to the YAGNI principle and won't do it this way.